src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -1868,5 +1884,14 @@ internal void EnableInvalidNameWarningState(string warningMessage)
             CurrentWarningMessage = warningMessage;
             IsWarningEnabled = true;
         }
+
+        private static MetadataLoadContext InitSharedPublishLoadContext()
+        {
+            // Retrieve the location of the assembly and the referenced assemblies used by the domain
+            var runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+            // Create PathAssemblyResolver that can resolve assemblies using the created list.
+            var resolver = new PathAssemblyResolver(runtimeAssemblies);
+             return new MetadataLoadContext(resolver);
```

Comment:
Why are we no longer doing this:
```
//and all the package assemblies.
            foreach (var assemblyFile in pkg.EnumerateAssemblyFilesInPackage())
            {
                assemblyPaths.Add(assemblyFile);
            }

            // Create PathAssemblyResolver that can resolve assemblies using the created list.
            var resolver = new PathAssemblyResolver(assemblyPaths);
            var mlc = new MetadataLoadContext(resolver);
```

Reply:
I found that it does not appear to be necessary to add the assemblies to the MLC as we pass the full path to the files we want to load.

Additionally, we only need the dependencies loaded if want to inspect those types - but we don't do much with the assemblies as far as I can tell except for use the location and assembly name.

In an MLC the dependencies are not required to load the assembly AFAIK.

Reply:
I do not see any immediate need to use those types...but as soon as we do ...we will have exceptions thrown. THese assemblies are not exposed publicly right ?

Reply:
no, and in addition this would have caused the same issue with ReflectionOnlyLoadContext.

Reply:
actually, the `PublicPackageViewModel.Assemblies` property is public and so is the class. These classes are only used in the UI layer though AFAIK and are not easily accessible from anywhere but our views, ie not from packages or extensions without clients hacking around.

Reply:
Can you explain why we are still not disposing the mlc instance (say on shutdown), now that it does not own references to assemblies in the package folder?

Reply:
So, I'm a bit confused after reading [this](https://learn.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext#create-metadataassemblyresolver-for-metadataloadcontext), just trying to understand it better. The docs seem to say that the resolver that is passed to the mlc resolves assemblies that are dependencies of the assembly being loaded (using the mlc). I can understand that this should include runtime assemblies, which most likely the assembly being loaded (for inspection) has a dependency on but won't it also have dependencies most likely on the other helper assemblies that are in the same package folder itself for the package that's being uploaded? Or are you saying that since the assembly being loaded and the other assemblies are in the same location, that isn't required? Maybe I'm missing something.

Reply:
yes, the binary we are loading very likely will have dependencies on other binaries in the package - but it doesn't matter. It does not matter because the MLC won't fail to load a binary simply because a dependency is not loaded.

It only becomes an issue when you try to inspect a type and that type has a dependency on assembly the resolver cannot resolve.

But in our case, as far as I can tell, we don't do any inspection, we just look at the assembly name and location... we're just using the MLC to do some sanity checking that the assembly is a real managed assembly.

ReflectionLoadContext also behaved like this.